secure/storage/vault/src/lib.rs

```diff
@@ -159,13 +172,174 @@ impl Client {
+            Err(response.into())
+        }
+    }
+
+    pub fn list_keys(&self) -> Result<Vec<String>, Error> {
+        let response = ureq::request("LIST", &format!("{}/v1/transit/keys", self.host))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
+            .call();
+        match response.status() {
+            200 => {
+                let list_keys: ListKeysResponse = serde_json::from_str(&response.into_string()?)?;
+                Ok(list_keys.data.keys)
+            }
+            404 => Err(Error::NotFound("transit/".into(), "keys".into())),
+            _ => Err(response.into()),
+        }
+    }
+
+    pub fn read_ed25519_key(
+        &self,
+        name: &str,
+    ) -> Result<Vec<ReadResponse<Ed25519PublicKey>>, Error> {
+        let response = ureq::get(&format!("{}/v1/transit/keys/{}", self.host, name))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
+            .call();
+        match response.status() {
+            200 => {
+                let read_key: ReadKeyResponse = serde_json::from_str(&response.into_string()?)?;
+                let mut read_response = Vec::new();
+                for (version, value) in read_key.data.keys {
+                    read_response.push(ReadResponse::new(
+                        value.creation_time,
+                        Ed25519PublicKey::try_from(base64::decode(&value.public_key)?.as_slice())?,
+                        version,
+                    ));
+                }
+                Ok(read_response)
+            }
+            404 => Err(Error::NotFound("transit/".into(), name.into())),
+            _ => Err(response.into()),
+        }
+    }
+
+    pub fn rotate_key(&self, name: &str) -> Result<(), Error> {
+        let response = ureq::post(&format!("{}/v1/transit/keys/{}/rotate", self.host, name))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
+            .call();
+        if response.ok() {
+            Ok(())
+        } else {
+            Err(response.into())
+        }
+    }
+
+    pub fn sign_ed25519(
+        &self,
+        name: &str,
+        data: &[u8],
+        version: Option<u32>,
+    ) -> Result<Ed25519Signature, Error> {
+        let data = if let Some(version) = version {
+            json!({ "input": base64::encode(&data), "key_version": version })
+        } else {
+            json!({ "input": base64::encode(&data) })
+        };
+
+        let response = ureq::post(&format!("{}/v1/transit/sign/{}", self.host, name))
+            .set("X-Vault-Token", &self.token)
+            .timeout_connect(10_000)
+            .send_json(data);
+        if response.ok() {
+            let signature: SignatureResponse = serde_json::from_str(&response.into_string()?)?;
+            let signature = &signature.data.signature;
+            let signature_pieces: Vec<_> = signature.split(':').collect();
+            let signature = signature_pieces
```

Comment:
Nit: Wow, it's kinda annoying that vault returns a signature string in this format (with each signature being prefixed with "vault:v1:"). Perhaps add a comment as to why we need to do this work, seeing as though it took me a while to figure out what was going on here (I had to parse this: https://www.vaultproject.io/api-docs/secret/transit/#sign-data)

Reply:
Aah nice! I missed that comment!